game.py

```diff
@@ -390,31 +415,42 @@ def _footer(self):
         return '</svg>\n'
 
     def read_best_time(self):
-        best_time = [180]
-        file_path = os.path.join(get_activity_root(), 'data', 'best-time')
+        best_time = [180, 180, 180, 180]
+        file_path = os.path.join(get_activity_root(), 'data', 'Best-time')
         if os.path.exists(file_path):
             with open(file_path, "r") as fp:
-                best_time = fp.readlines()
-        return int(best_time[0])
+                best_time = fp.readline()
+        return best_time
 
     def save_best_time(self):
-        file_path = os.path.join(get_activity_root(), 'data', 'best-time')
-        int_best_time = self.read_best_time()
-
-        if not int_best_time <= self.elapsed_time:
-            int_best_time = self.elapsed_time
+        file_path = os.path.join(get_activity_root(), 'data', 'Best-time')
+        best_time = self.read_best_time()
+        if type(best_time) is str:
+            best_time = self.convert_to_int_list(best_time)
+        if not best_time[self._edge-4] <= self.elapsed_time:
+            best_time[self._edge-4] = self.elapsed_time
         with open(file_path, "w") as fp:
-            fp.write(str(int_best_time))
+            fp.write(str(best_time))
 
     def load_best_time(self):
         best_time = self.read_best_time()
+        if type(best_time) is str:
+            best_time = self.convert_to_int_list(best_time)
         try:
             return best_time
         except (ValueError, IndexError) as e:
             logging.exception(e)
             return 0
         return 0
 
+    def convert_to_int_list(self, str_data):
+        list_str = str_data
+        list_int = []
+        temp = ((list_str.split("]"))[0].split("["))[1].split(", ")
+        for i in range(0, len(temp)):
+            list_int.append(int(temp[i]))
+        return list_int
+
```

Comment:
I think that this can be better done as there's lots of assignments that can be avoided, the `list_str` seems to be redundant as you can achieve the same purpose while using a slice of `str_data` the splitting done in `temp` is unreadable, you can break the split down so as to make it readable.

When looping through a range, if you want to start from the first item `0` there's no need to put `0` in the loop as the range starts from there except stated otherwise so you can just have `for i in range(len(temp)):`.